src/blob/core.rs

```diff
@@ -183,31 +183,54 @@ impl Blob {
         debug!("blob read any");
         let entry = self
             .get_any_entry(key, meta)
-            .await?
+            .await
+            .unwrap()
             .ok_or_else(|| Error::from(ErrorKind::RecordNotFound))?;
         debug!("blob read any entry found");
         let buf = entry
             .load()
             .await
-            .with_context(|| format!("failed to read key {:?} with meta {:?}", key, meta))?
+            .with_context(|| format!("failed to read key {:?} with meta {:?}", key, meta))
+            .unwrap()
             .into_data();
         debug!("blob read any entry loaded bytes: {}", buf.len());
         Ok(buf)
     }
 
     #[inline]
-    pub(crate) fn read_all<'a>(&'a self, key: &'a [u8]) -> Entries<'a> {
-        self.index.get_entries(key, self.file.clone())
+    pub(crate) async fn read_all(&self, key: &[u8]) -> Result<Option<Vec<Entry>>> {
```

Comment:
should we rename it to `get_all_entries`

Reply:
I don't know, we don't have other plans for `read_all` fn name